src/data/polynomial.lean

```diff
@@ -238,18 +228,17 @@ variables (R) (A) [semiring A] [algebra R A] [semiring B] [algebra R B] {x : A}
 
 /-- Given a valuation `x` of the variable in an `R`-algebra `A`, `aeval R A x` is
 the unique `R`-algebra homomorphism from `R[X]` to `A` sending `X` to `x`. -/
-def aeval : A ≃ (polynomial R →ₐ[R] A) :=
-(powers_hom A).trans (monoid_algebra.lift R (multiplicative ℕ) A)
+def aeval : A ≃ (polynomial R →ₐ[R] A) := (powers_hom A).trans (add_monoid_algebra.lift R ℕ A)
 
 variables {R A}
 
-theorem aeval_def' :
-  aeval R A x p = monoid_algebra.lift R (multiplicative ℕ) A (powers_hom A x) p := rfl
+theorem aeval_def' : aeval R A x p = add_monoid_algebra.lift R ℕ A (powers_hom A x) p := rfl
```

Comment:
The right hand side looks quite intimidating compared to `p.eval x`. Is this the new simp-normal-form?

Reply:
In general there seem to be very few simp-lemmas for `aeval`. I would expect simp-lemmas for `aeval_X` and `aeval_C`.

Reply:
This is not a `simp` lemma. This is a lemma “show me an ugly definition, I want to apply some theorem about `add_monoid_algebra.lift`”. I'll see if it can be removed. You can find `aeval_X` and `aeval_C` below.

Reply:
Ok, thanks for explaining. It certainly can stay.